MaxText/pyconfig.py

```diff
@@ -187,6 +187,8 @@ def validate_no_keys_overwritten_twice(keys1: list[str], keys2: list[str]):
 def _lists_to_tuples(l: list[Any]) -> Union[tuple[Any], list[Any]]:
   return tuple(_lists_to_tuples(x) for x in l) if isinstance(l, list) else l
 
+def _str_to_list(s: str) -> list[Any]:
```

Comment:
my recommendation of using "device" "host" "remat" would require slightly more parsing, looping thorugh the config fields